### DISCUSSION TITLE
Add template_fields as a parameter in BaseOperator

### DISCUSSION THREAD

**Discussion by andresitorusz** [apache/airflow#32396] [Q&A] [answered]
Hi everyone, what if we add template_fields as a parameter in **BaseOperator**?

The goal is that we can add `template_fields` according to our needs in operators that extend **BaseOperator**.

I thought about asking this, because right now I'm trying to send a query with the Jinja2 template included to a **SparkKubernetesOperator** task. I'm confused about how my query will be rendered, because only the `application_file`  and `namespace` are included in the `template_fields` in the **SparkKubernetesOperator**.

Currently, I'm sending the query via parameter `params`. And it looks like this in the Airflow Webserver UI:

![image](https://github.com/apache/airflow/assets/36465150/0a5fcdbf-536d-4624-ac14-6972767bb956)

Kindly help if there is a solution to my problem. Thanks in advance!

**Comment by potiuk** [comment, marked as answer]
You can extend SparkKubernetesOperator:

```
class MySKO(SparkKubernetesOperator):
     template_fields = .....
```
and use your new operator.

And yeah if you think overriding template_fields from the constructor is simpler (I am no sure) - feel free to propose PR for that.

**Reply by potiuk** [reply]
Having a custom operator though seems simpler and more reusable.

**Reply by andresitorusz** [reply]
Thanks for the suggestion. I was able to solve my problem (sending the templated query to SparkKubernetesOperator) by writing the arguments to the application_file YAML, like this:
```
with open(f'{PYTHONPATH}/{RDBMS_TO_BQ_APPLICATION_FILE}') as f:
        application_file = yaml.safe_load(f)

application_file['spec']['arguments'] = [
        f"--extract_query={self.__generate_extract_query(schema=schema)}",
        ...
]

spark_kubernetes_operator_task = SparkKubernetesOperator(
        application_file = yaml.safe_dump(application_file),
        ...    
)
```

**Reply by potiuk** [reply]
FYI: I think in this case it will only work during parsing time of the DAG, not during execution of the task, so if your parameters might be different between runs, that might not be the best idea.

**Reply by andresitorusz** [reply]
In my case, there are no differences in the parameters between runs, except the `{{ data_interval_start }}` and `{{ data_interval_end }}` templates in the extract_query. Other than that, is just parameters from the DAG (and the task) config.

**Reply by potiuk** [reply]
should be good then

**Reply by andresitorusz** [reply]
Thank you!
